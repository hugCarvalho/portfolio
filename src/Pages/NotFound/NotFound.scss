.NotFound {
  color: rgb(202, 226, 230);
  padding: 40px;
  font-size: 1.6rem;
  text-align: center;
  line-height: 2.7;

  button {
    background-color: rgb(34, 33, 33);
    color: white;
    min-width: 103px;
    width: 105px;
    margin: 20px 0;
    padding: 4px;
    opacity: 0.9;
    border-radius: 3px;

    &:hover {
      opacity: 1;
    }
  }
}
