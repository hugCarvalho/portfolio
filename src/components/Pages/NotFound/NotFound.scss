.not-found {
  font-size: 1.6rem;
  color: rgb(202, 226, 230);
  padding: 40px;
  text-align: center;
  line-height: 6.5rem;

  button {
    background-color: rgb(34, 33, 33);
    color: white;
    padding: 4px;
    margin: 20px 0;
    border-radius: 3px;
    opacity: 0.9;
    min-width: 103px;
    width: 105px;

    &:hover {
      opacity: 1;
    }
  }
}
