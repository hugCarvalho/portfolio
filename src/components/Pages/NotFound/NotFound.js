import React from "react";
import "./NotFound.scss";
import { Link } from "react-router-dom";

function NotFound() {
  return (
    <main className="NotFound">
      <h2>
        <p>An error has ocurred! </p>
        <p>
          Please click the button below to return to the home page or use the
          above navigation buttons to return to the website
        </p>

        <Link to="/projects">
          <button>Back to Home Page</button>
        </Link>
      </h2>
    </main>
  );
}

export default NotFound;
