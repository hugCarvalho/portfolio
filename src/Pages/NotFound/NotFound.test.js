import React from "react";
import { render, screen } from "@testing-library/react";
import NotFound from "./NotFound";
import { MemoryRouter } from "react-router-dom";

describe("NotFound renders correctly", () => {
  test("button back to home page exists", () => {
    render(
      <MemoryRouter>
        <NotFound />
      </MemoryRouter>
    );

    const button = screen.getByRole("link", { name: /back to home page/i });
    expect(button.href).toContain("projects");
  });
  test("should match snapshot", () => {
    const { container } = render(
      <MemoryRouter>
        <NotFound />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
