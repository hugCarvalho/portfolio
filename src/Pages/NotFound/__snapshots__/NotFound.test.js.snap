// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound renders correctly should match snapshot 1`] = `
<div>
  <main
    class="NotFound"
  >
    <h2>
      <p>
        An error has ocurred! 
      </p>
      <p>
        Please click the button below to return to the home page or use the above navigation buttons to return to the website
      </p>
      <a
        href="/projects"
      >
        <button>
          Back to Home Page
        </button>
      </a>
    </h2>
  </main>
</div>
`;
